//#service-request-reply
syntax = "proto3";

// The file handler service definition.
service GraphPainterService {
  // Sends a greeting
  rpc GreetGP (SayHelloGPMessage) returns (SayHelloGPMessage) {}

  rpc PaintCandlestick (ChartRequest) returns (ChartResponse) {}

  rpc PaintChart (ChartRequest) returns (ChartResponse) {}

}

message ChartRequest {
  string datas = 1;
  string tokenInfo = 2;
  string options = 3;
}

message ChartResponse {
  bytes image = 1;
}

message SayHelloGPMessage {
  string message = 1;
}

